detectors/aws/eks/eks_resource_detector.go

```diff
@@ -0,0 +1,242 @@
+	}
+
+	// Set HTTP request header with authentication credentials
+	authHeader, err := getK8sCredHeader()
+	if err != nil {
+		return "", err
+	}
+	request.Header.Set("Authorization", authHeader)
+
+	// Get certificate
+	caCert, err := ioutil.ReadFile(k8sCertPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read file with path %s", k8sCertPath)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	// Set HTTP request timeout and add certificate
+	client := &http.Client{
+		Timeout: timeoutMillis * time.Millisecond,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}
+
+	// Execute HTTP request
+	response, err := client.Do(request)
+	if err != nil || response.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to execute HTTP request with method=%s, URL=%s, Status Code=%d: %w", httpMethod, URL, response.StatusCode, err)
+	}
+
+	// Retrieve response body from HTTP request
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read response from HTTP request with method=%s, URL=%s: %w", httpMethod, URL, err)
+	}
+
+	return string(body), nil
+}
+
+// getK8sCredHeader retrieves the kubernetes credential information
+func getK8sCredHeader() (string, error) {
+	content, err := ioutil.ReadFile(k8sTokenPath)
+	if err != nil {
+		return "", fmt.Errorf("getK8sCredHeader() error: cannot read file with path %s", k8sTokenPath)
+	}
+
+	return "Bearer " + string(content), nil
+}
+
+// getClusterName retrieves the clusterName resource attribute
+func getClusterName(utils detectorUtils) (string, error) {
+	resp, err := utils.fetchString("GET", k8sSvcURL+cwConfigmapPath)
+	if err != nil {
+		return "", fmt.Errorf("getClusterName() error: %w", err)
+	}
+
+	// parse JSON object returned from HTTP request
+	var respmap map[string]json.RawMessage
+	err = json.Unmarshal([]byte(resp), &respmap)
+	if err != nil {
+		return "", fmt.Errorf("getClusterName() error: cannot parse JSON: %w", err)
+	}
+	var d data
+	err = json.Unmarshal(respmap["data"], &d)
+	if err != nil {
+		return "", fmt.Errorf("getClusterName() error: cannot parse JSON: %w", err)
+	}
+
+	clusterName := d.ClusterName
+
+	return clusterName, nil
+}
+
+// getContainerID retrieves the containerID resource attribute
+// this function implements the detectorUtils interface
+func (eksUtils eksDetectorUtils) getContainerID() (string, error) {
+
+	// Read file
+	fileData, err := ioutil.ReadFile(defaultCgroupPath)
+	if err != nil {
+		return "", fmt.Errorf("getContainerID() error: cannot read file with path %s: %w", defaultCgroupPath, err)
+	}
+
+	// Retrieve containerID from file
+	splitData := strings.Split(strings.TrimSpace(string(fileData)), "\n")
+	for _, str := range splitData {
+		if len(str) > containerIDLength {
+			return str[len(str)-containerIDLength:], nil
+		}
+	}
```

Comment:
This seems like an error prone algorithm to determining the container ID. If I understand this correctly it is looking for a line longer than some length and then assuming that line ends with the container ID?

Based on this:

```
docker run --rm debian cat /proc/self/cgroup
12:cpu,cpuacct:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
11:devices:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
10:pids:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
9:cpuset:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
8:net_cls,net_prio:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
7:memory:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
6:hugetlb:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
5:freezer:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
4:rdma:/
3:blkio:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
2:perf_event:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
1:name=systemd:/docker/422a3caf71aecd248eb39e2d6e2b4490992a7703e995745ca0ba4ae989f73e88
0::/system.slice/containerd.service
```

Would it make more sense to look for lines that match `^.*/docker/(.+)$` instead? That or better document our assumptions as to why we think length is appropriate to match on.

Reply:
Yep that would make more sense! I have changed the implementation to use regex instead. Thanks!